node/src/types/node_config.rs

```diff
@@ -75,6 +75,9 @@ pub struct NodeConfig {
 
     /// If true, prevents a node from shutting down if it is supposed to be a validator in the era.
     pub prevent_validator_shutdown: bool,
+
+    /// Flag which forces node to shut down after initial sync is done
+    pub catch_up_and_shutdown: bool,
```

Comment:
I think it would be more correct to add a `CompleteBlock` variant to `SyncHandling` enum to manage how the node handles this synching behavior.